lib/measured/unit.rb

```diff
@@ -0,0 +1,59 @@
+class Measured::Unit
+  include Comparable
+
+  def initialize(name, aliases: [], value: nil)
+    @name = name.to_s
+    @names = ([@name] + aliases.map{|n| n.to_s }).sort
+
+    @conversion_amount, @conversion_unit = parse_value(value) if value
+  end
+
+  attr_reader :name, :names, :conversion_amount, :conversion_unit
+
+  def to_s
+    if conversion_string
+      "#{ @name } (#{ conversion_string })"
+    else
+      @name
+    end
+  end
+
+  def inspect
+    "#<Measured::Unit: #{ @name } (#{ @names.join(", ") }) #{ conversion_string }>"
+  end
+
+  def <=>(other)
+    if self.class == other.class
+      if other.names != @names
+        other.names <=> @names
+      else
+        other.conversion_amount <=> @conversion_amount
+      end
+    else
+      @name <=> other
+    end
+  end
+
+  def inverse_conversion_amount
+    if conversion_amount.is_a?(Rational)
+      Rational(conversion_amount.denominator, conversion_amount.numerator)
```

Comment:
I did make this change to prevent `Rational` from getting prematurely casted to a `BigDecimal`, though in the calls to multiply the value by the factor it is always eventually returned as `BigDecimal`. But this is slightly better.